### ISSUE TITLE
Flatpak build for Linux (in progress)

### ISSUE THREAD

**Issue by slembcke** [Aleph-One-Marathon/alephone#344]
I have the basics of building flatpak versions of the original Trilogy here:
https://github.com/slembcke/flatpak-marathon

The `build.sh` script will build .flatpak files for each of the original games in the trilogy. A few people on the GamingOnLinux discord have tested them, and they seem to work. I have someone that said they would try them on the Steam Deck, but they aren't in a hurry. hehe

So I guess there are multiple paths from here:
* The flatpaks could be hosted directly on the lhowon site as binaries for the Linux version. (The source version will always appeal to this crowd too I suppose)
* Someone could submit them to Flathub. I don't mind managing this if that's kosher? It looks like the process is pretty painless compared to consoles, Google or Apple. (I've never done the MS store, but my experience as a user was so bad I can't imagine why I would bother...)

Next steps:
* Preferences: It looks like the Mac/Win builds ship with modified preferences to use the XBLA HUD by default and some other changes maybe? On my machines, the default mouse sensitivity of 1.0 is *way* too sensitive. I need to verify this with others though.
* Launcher: For licensing reasons, I could make the app into an Aleph One launcher with links to the lhowon site, level guides, install data for or launch the Marathon trilogy, HD texture packs, or some of the total conversions. (RubiconX was so good!) Depending on how it's implemented, this could be useful for the Mac/Windows versions too I suppose?

What direction would you like to take this?

**Comment by pizzadude** [comment]
There exists this: https://gitlab.com/elagost/alephone-marathon-launcher

Maybe you could just make a flatpak of the launcher and the game binary itself?

**Comment by slembcke** [comment]
Yeah, that's not too far off from what I had in mind really. Maybe a few extra icons, and some links to the project page, level guides, etc.

**Comment by ribbybbir** [comment]
It's great that a flatpak is being worked on for the Aleph One source port of the Marathon series. Missed the ambience of the map environments. Almost like Myst. I wonder why it has been gaining a lack of interest or it's just me who skipped out. I can't access the traxus hosted mods anymore. Simpli7y site has been occasionally going down so Marathon mods seem to be a rare thing. I want to have the mods available in ready access, common archive file formats whenever possible. archive.org or github.com sounds like a nice start. I can't have .hqx or .sit blocking the way either.

By the way, I have forgotten about how to load custom mods to the Aleph One source port. Anyone still remember? I am sure there's a CLI and GUI way.

**Comment by treellama** [comment]
@slembcke May I include these in the main repo (with modifications) under the GPL 3?

**Comment by slembcke** [comment]
Yes, definitely.

I haven't touched these for a few months because I got busy with work, but IIRC it was basically ready to go except for the licensing issue. As a workaround for that I had been intending to make some sort of little launcher that could download the game data from Aleph servers, but never got around to it. Having it in the main repo kinda just solves that I guess.

**Comment by elagost** [comment]
Hi all! I've made a few changes to the flatpak (https://github.com/slembcke/flatpak-marathon/pull/1) and am hosting it currently. It installs and runs. There are install URLs on  [my web page for the flatpak](https://elagost.com/flatpak/alephone.html). This is a proof of concept and ideally it'd be submitted to flathub at some point.

@treellama, would you be interested in having this submitted to flathub? [It looks fairly easy](https://github.com/flathub/flathub/wiki/App-Submission#how-to-submit-an-app) but if you would rather not, I'd gladly do it.

As far as the launcher - it was a messy hobby project, but it's in a better state now [over here](https://git.sr.ht/~elagost/alephone-launcher) - no more calling out to a separate shell script! I think that packaging 3 separate flatpaks with their own game data and .desktop files is more user friendly but from the game data README it might be bad form to distribute the game data from flathub.

**Comment by slembcke** [comment]
Ah, I almost forgot, but another option I ran across is to use Flatpak's "extra data" feature. https://docs.flatpak.org/en/latest/conventions.html#exporting-through-extra-data It downloads files during installation from an external server, which seems like it would resolve the legal fuzzyness involved. You could pass off the effort of maintaining the Flathub version while still satisfying the distribution agreement?

**Comment by treellama** [comment]
For now, my intent is to distribute them from GitHub. Flathub is a possibility, I will look into that after the next release.

**Comment by treellama** [comment]
b523e38b

**Comment by darealshinji** [comment]
Inspired by the flatpak release I had written a build script to create AppImages:
https://github.com/darealshinji/alephone/tree/appimage/appimage

Four different bundles are created:
one that only contains AlephOne ~20 MB,
one for each Marathon game, ~50 MB each,
one with all three games and a graphical selection (using kdialog/zenity/xmessage), ~105 MB

To reduce the bundle size FFmpeg is build from source, so we don't bundle countless thirdparty libraries.

**Comment by treellama** [comment]
We could consider adding an app image. With the desktop files, I don't see the need for a launcher.

Grabbing ffmpeg from git without a tag is risky, since their API is not very stable, and could break the build.

**Comment by darealshinji** [comment]
> We could consider adding an app image. With the desktop files, I don't see the need for a launcher.

I have removed the "trilogy" build.

> Grabbing ffmpeg from git without a tag is risky, since their API is not very stable, and could break the build.

The build file will now checkout release 6.0.

**Comment by Kekun** [comment]
Hi! Just a random Linux user/dev who got curious about AlephOne and Marathon. My first reflex has been to search for it on Flathub, then I found your website and its flatpaks: thanks a lot for puclishing them! But then I wondered, of there are flatpaks, why not distribute them on Flathub? So I ended up here. I'm glad to read distributing them there is a goal. :)

I published some apps and games there a few years ago, maybe I can help submitting the Marathon games there?

**Comment by fpiesche** [comment]
> Hi! Just a random Linux user/dev who got curious about AlephOne and Marathon. My first reflex has been to search for it on Flathub, then I found your website and its flatpaks: thanks a lot for puclishing them! But then I wondered, of there are flatpaks, why not distribute them on Flathub? So I ended up here. I'm glad to read distributing them there is a goal. :)
> 
> I published some apps and games there a few years ago, maybe I can help submitting the Marathon games there?

Similarly I've been submitting various open-source games projects to Flathub recently, somewhat selfishly because I've got a Steam Deck over here :D I've been using the Flatpak build of Aleph One for a while now and it works fine - as with Adrien above I'd be happy to help get A1 published on Flathub (and all the relevant data is already there so the work needed should be pretty minimal!). The A1 Flatpak alone, without the content included, should be very uncomplicated in terms of rights shouldn't it?

**Comment by treellama** [comment]
We are not ready to keep the flatpaks updated on flathub at this time. Perhaps some day!

**Comment by fpiesche** [comment]
The Flatpaks on Flathub will actually update themselves - Flathub has its own CI infrastructure. Submitting A1 to Flathub would involve forking the flathub/flathub repo, adding the .yml build manifest over there and creating a PR from the fork to the Flathub org.

After this is merged, any contributors here will be added with write permissions to the new `flathub/org.lhowon.alephone` (or something along those lines) repository and Flathub's build infra will monitor this repository for new tags on a daily basis. When a new tag matching a regex (specified in the Flatpak manifest) is created, Flathub's infra will automatically create a PR on the `flathub/org.lhowon.alephone` repo to update the .yml file with the new tag, run a test build and once the PR is merged will update the Aleph One build on the public Flathub repository - at which point users will see the updated version when they run `flatpak update` or when their desktop integration checks Flathub for updates :)
